src/js/elements/menu.js

```diff
@@ -1,174 +1,176 @@
 const loadingClsName = 'data-mapper-content__loading';
 
-function subindicatorsInCategory(category) {
-    let count = 0;
-    let subcategories = Object.values(category.subcategories);
-    for (const idx in subcategories) {
-        let subcategory = subcategories[idx];
-        count += subindicatorsInSubCategory(subcategory);
-    }
-
-    return count;
-}
 
 function subindicatorsInSubCategory(subcategory) {
 
-    let count = 0;
-    const indicators = Object.values(subcategory.indicators);
-    if (indicators.length > 0) {
-        for (const idx in indicators) {
-            let indicator = indicators[idx];
-            count += subindicatorsInIndicator(indicator);
-        }
+  let count = 0;
+  const indicators = Object.values(subcategory.indicators);
+  if (indicators.length > 0) {
+    for (const idx in indicators) {
+      let indicator = indicators[idx];
+      count += subindicatorsInIndicator(indicator);
     }
+  }
 
-    return count;
+  return count;
 }
 
+// TODO: possible provide the simplest possible arguments to methods
 function subindicatorsInIndicator(indicator) {
-    return indicator.metadata.groups.length;
+  return indicator.metadata.groups.length;
 }
 
 // TODO this entire file needs to be refactored to use thhe observer pattern
-export function loadMenu(dataMapperMenu, data, subindicatorCallback) {
-    function addSubIndicators(wrapper, category, subcategory, indicator, groups, indicators, choropleth_method, indicatorId) {
-
-        $(".data-category__h3", wrapper).remove();
-        $(".data-category__h4", wrapper).remove();
-
-        if (groups !== null && typeof groups.subindicators !== 'undefined') {
-            groups.subindicators.forEach((subindicator) => {
-                const newSubIndicatorElement = indicatorItemTemplate.cloneNode(true);
-                $(".truncate", newSubIndicatorElement).text(subindicator);
-                $(newSubIndicatorElement).attr('title', subindicator);
-
-                wrapper.append(newSubIndicatorElement);
-
-                const parents = {
-                    category: category,
-                    subcategory: subcategory,
-                    indicator: indicator
-                }
-
-                $(newSubIndicatorElement).on("click", (el) => {
-                    setActive(el);
-                    if (subindicatorCallback != undefined)
-                        subindicatorCallback({
-                            el: el,
-                            data: data,
-                            indicatorTitle: indicator,
-                            selectedSubindicator: subindicator,
-                            indicators: indicators,
-                            parents: parents,
-                            choropleth_method: choropleth_method,
-                            indicatorId: indicatorId
-                        })
-                });
-            });
+export class LoadMenu extends Controller {
```

Comment:
Why extend the Controller class rather than Component?

Reply:
If you intended to extend Component, then you'll need to pass in the parent into the constructor and call super(parent)

Reply:
Done. I used the DataMapperMenu stub component you had created.